### PR TITLE
Add support for Stockport, UK

### PR DESCRIPTION
**Pull request by MuddyRock**
I may not have created the pull-request correctly - apologies if that's the case. I'm still a bit new to GitHub and Python.
There is probably a better way of iterating through the results than the one I went for. See above. Any feedback welcome.

### CODE REVIEW

**Comment by MuddyRock** [comment]
This requires a Unique Property Reference Number (uprn) to determine the URL to use.
You can find your UPRN by going to 
https://www.stockport.gov.uk/find-your-collection-day
and entering your postcode (if you're testing this, and you live outside Stockport, try SK6 3AA), then picking your house number.
The UPRN is the final part of the URL that you are then directed to e.g.
for SK6 3AA, house number 25, you are directed to 
https://myaccount.stockport.gov.uk/bin-collections/show/10090543805
The UPRN is 10090543805

File: custom_components/waste_collection_schedule/waste_collection_schedule/source/stockport_uk.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+class Source:
+    def __init__(
+        self, uprn: int = None
+    ):
+        self._uprn = uprn
+        if not self._uprn:
+            _LOGGER.error(
+                "uprn must be provided in config"
+            )
+        self._session = requests.Session()
```
Comment (mampfes) [inline custom_components/waste_collection_schedule/waste_collection_schedule/source/stockport_uk.py:37]: Don't create session in __init__, but in fetch.

File: custom_components/waste_collection_schedule/waste_collection_schedule/source/stockport_uk.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            bin_date_pos = bins.find("<p>",bin_name_start_position)
+            bin_date_exc_day_of_week_pos = bins.find(", ",bin_date_pos)+2
+            bin_date_end_pos = bins.find("</p>",bin_date_exc_day_of_week_pos)
+            bin_date_string = bins[bin_date_exc_day_of_week_pos:bin_date_end_pos]
+            bin_date = datetime.strptime(bin_date_string,'%d %B %Y').date()
+            entries.append(
+                Collection(
+                    date=bin_date,
+                    t=bin_name,
+                    icon=ICONS[bin_name],
```
Comment (mampfes) [inline custom_components/waste_collection_schedule/waste_collection_schedule/source/stockport_uk.py:69]: `icon=ICONS.get(bin_name)`
This variant doesn't crash if the bin name is missing in the dict, but fall back to default.

**Review by mampfes — CHANGES_REQUESTED** [review]
- Please rename source to stockport_gov_uk
- Please add corresponding documentation as markdown file
- Please add links to new source in README.md and info.md
--> https://github.com/mampfes/hacs_waste_collection_schedule#how-to-add-new-sources

**Comment by MuddyRock** [comment]
All requested changes made.
Specifically:
- Moved  create session to fetch
- Used correct method to check for icons
- Updated info.md
- Updated README.md
- renamed file to stockport_gov_uk.py
- Added doc file stockport_gov_uk.md in doc\source

I can't see how to re-request approval from you in GitHub for the changes.
Do I create a new pull request?

**Comment by mampfes** [comment]
The diff for this PR shows major changes in README.md and info.md - maybe a problem with line endings. Check your editor and update README.md and info.md so that is shows only the newly added lines.

**Comment by mampfes** [comment]
I did the requested changes in #622

**Comment by mampfes** [comment]
Thanks for your contribution.

**Pull request merged**